table/block_based_table_reader.cc

```diff
@@ -1436,23 +1504,20 @@ BlockBasedTable::BlockEntryIteratorState::NewSecondaryIterator(
   BlockHandle handle;
   Slice input = index_value;
   Status s = handle.DecodeFrom(&input);
-  auto iter = NewDataBlockIterator(table_->rep_, read_options_, handle, nullptr,
-                                   is_index_, s);
-  if (block_cache_cleaner_) {
-    uint64_t offset = handle.offset();
-    {
-      ReadLock rl(&cleaner_mu);
-      if (cleaner_set.find(offset) != cleaner_set.end()) {
-        // already have a reference to the block cache objects
-        return iter;
-      }
+  auto rep = table_->rep_;
+  if (block_map_) {
+    auto block = block_map_->find(handle.offset());
+    assert(block != block_map_->end());
+    if (LIKELY(block != block_map_->end())) {
```

Comment:
yes, but nobody is actively looking at logs for failures. This is not a place with visibility. If you don't feel comfortable with move it, I suggest we call abort() or something like that here.